kernel/src/actions/mod.rs

```diff
@@ -683,11 +683,96 @@ pub(crate) struct DomainMetadata {
     removed: bool,
 }
 
+/// Though technically not an action, we include the CRC (version checksum) file here. A [CRC file]
```

Comment:
I think I might prefer this in its own `crc.rs` module? This file is already pretty huge, so i'll take any reason to not add more to it :)

Reply:
yea I thought that as I wrote this hahaha - will split out!